### ISSUE TITLE
Status bar in a task application not visible unless window is resized

### ISSUE THREAD

**Issue by jenniportman** [enthought/pyface#329]
(With pyface 6.0.0-1, qt4, python 3)
I have added a `status_bar` to a pyface `Task` however by default it is not visible until the user resizes the window vertically. This can be seen by adding a line like `status_bar = StatusBarManager(message="Hello World!")` to the example task in the advanced tasks examples.

Here is what the window looks like in the example (I have closed the dock pane as not relevant):
<img width="912" alt="screen shot 2018-08-01 at 16 12 25" src="https://user-images.githubusercontent.com/4663211/43530698-1dd4863e-95a6-11e8-8a18-9227ecc8b550.png">

Here is what the window looks like after resizing (note the status bar at the bottom, which now stays visible regardless of resizing):
<img width="936" alt="screen shot 2018-08-01 at 16 16 37" src="https://user-images.githubusercontent.com/4663211/43530782-4d89d078-95a6-11e8-983d-d584b1ad15ff.png">

**Comment by jenniportman** [comment]
Taking the status bar example a bit further, if I add another task which also has a status bar and flip back and forth between the two, I get the following error when going back to the first task:
```
2018-08-02 10:12:52 ERROR    [traits:238] Exception occurred in traits notification handler for object: <pyface.tasks.task_window.TaskWindow object at 0x13a882e08>, trait: status_bar_manager, old value: <pyface.ui.qt4.action.status_bar_manager.StatusBarManager object at 0x12f44c150>, new value: <pyface.ui.qt4.action.status_bar_manager.StatusBarManager object at 0x13a8abd58>
Traceback (most recent call last):
  File "/Users/jportman/.edm/envs/test_pyface/lib/python3.6/site-packages/traits/trait_notifiers.py", line 340, in __call__
    self.handler( *args )
  File "/Users/jportman/.edm/envs/test_pyface/lib/python3.6/site-packages/pyface/ui/qt4/application_window.py", line 163, in _status_bar_manager_changed
    self._create_status_bar(self.control)
  File "/Users/jportman/.edm/envs/test_pyface/lib/python3.6/site-packages/pyface/ui/qt4/application_window.py", line 76, in _create_status_bar
    self.control.setStatusBar(status_bar)
RuntimeError: wrapped C/C++ object of type QStatusBar has been deleted
```
I suspect that the error might be due to an incorrect initialization of the status bar manager (as also suggested by @jwiggins in a discussion we had about this)

**Comment by corranwebster** [comment]
Looking at things right now, and it seems that reliability could be improved by explicitly deleting and recreating the status bar control when adding/removing a status bar from a window.

For the non-appearance, I have a sneaking suspicion it is that `TaskWindow` is overriding something in the base `ApplicationWindow` but haven't sorted the control flow yet in my head. I have verified that it appears as expected in the base `ApplicationWindow` class.

**Comment by jenniportman** [comment]
Minimal code which shows the two issues:
```
from pyface.action.api import StatusBarManager
from pyface.api import GUI
from pyface.tasks.action.api import SMenuBar, SMenu, TaskToggleGroup
from pyface.tasks.api import EditorAreaPane, Task, TaskWindow


class ExampleTask(Task):
    id = 'example_task'
    name = 'Example'

    def _status_bar_default(self):
        return StatusBarManager()

    def _menu_bar_default(self):
        menu_bar = SMenuBar(SMenu(TaskToggleGroup(), id='View', name='&View'))
        return menu_bar

    def create_central_pane(self):
        return EditorAreaPane()

    def create_dock_panes(self):
        return []


class SecondExampleTask(Task):
    id = 'second_example_task'
    name = 'Second Example'

    def _status_bar_default(self):
        return StatusBarManager()

    def _menu_bar_default(self):
        menu_bar = SMenuBar(SMenu(TaskToggleGroup(), id='View', name='&View'))
        return menu_bar

    def create_central_pane(self):
        return EditorAreaPane()

    def create_dock_panes(self):
        return []


def main():
    gui = GUI()
    window = TaskWindow(size=(800, 600))

    task = ExampleTask()
    window.add_task(task)

    other_task = SecondExampleTask()
    window.add_task(other_task)

    window.open()
    gui.start_event_loop()


if __name__ == '__main__':
    main()
```

Steps:
- Run code, notice that no status bar is shown (window looks like this for me)
<img width="912" alt="screen shot 2018-08-03 at 09 12 03" src="https://user-images.githubusercontent.com/4663211/43632283-8cefce54-96fd-11e8-9ea6-47ae53f507ca.png">

- Switch to second task from the menu bar View -> Second Example, switch back to first task (View -> Example)
- See error in terminal:
```
Exception occurred in traits notification handler for object: <pyface.tasks.task_window.TaskWindow object at 0x125c86f68>, trait: status_bar_manager, old value: <pyface.ui.qt4.action.status_bar_manager.StatusBarManager object at 0x125cc8f68>, new value: <pyface.ui.qt4.action.status_bar_manager.StatusBarManager object at 0x125cc88e0>
Traceback (most recent call last):
  File "/Users/jportman/.edm/envs/test_pyface/lib/python3.6/site-packages/traits/trait_notifiers.py", line 340, in __call__
    self.handler( *args )
  File "/Users/jportman/Projects/pyface/build/lib/pyface/ui/qt4/application_window.py", line 163, in _status_bar_manager_changed
    self._create_status_bar(self.control)
  File "/Users/jportman/Projects/pyface/build/lib/pyface/ui/qt4/application_window.py", line 76, in _create_status_bar
    self.control.setStatusBar(status_bar)
RuntimeError: wrapped C/C++ object of type QStatusBar has been deleted
```

The status bar becomes visible if the window is resized vertically and stays visible once that is done (but the error above is still raised regardless of doing that).

**Comment by corranwebster** [comment]
I can't replicate the status bar not appearing, but I _can_ replicate the error when switching tasks.

**Comment by corranwebster** [comment]
And I have a potential fix, which is to explicitly set the old `StatusBarManager`'s `status_bar` to None in `_status_bar_manager_changed`.

So short-term workaround before there is a new release is to use a subclass with the following method:
```
    def _status_bar_manager_changed(self, old, new):
        if old is not None:
            old.status_bar = None
        if self.control is not None:
            self._create_status_bar(self.control)
```
